### PR TITLE
[MRG] FIX: read easycap-M10 montage

### PR DESCRIPTION
**Pull request by christianbrodbeck**
There was a bug in the standard montage reader that read easycap-M1 even when `easycap-M10` was specified.

While investigating I also noticed that the `DigMontage.plot` default parameters don't agree with the documentation, below I changed the default parameters to match the docs (since the docs do agree with the default parameters of `plot_montage()`).

### CODE REVIEW

File: mne/channels/_standard_montage_utils.py [hunk 1]
```diff
@@ -50,13 +50,16 @@ def _easycap(basename, head_size):
...
-                   ch_pos['TP10']], axis=0)
-    rpa *= head_size / np.linalg.norm(rpa)
+    if basename == 'easycap-M1.txt':
+        nasion = np.concatenate([[0], ch_pos['Fpz'][1:]])
+        lpa = np.mean([ch_pos['FT9'], ch_pos['TP9']], axis=0)
+        lpa *= head_size / np.linalg.norm(lpa)  # on sphere
+        rpa = np.mean([ch_pos['FT10'], ch_pos['TP10']], axis=0)
+        rpa *= head_size / np.linalg.norm(rpa)
+    elif basename == 'easycap-M10.txt':
+        nasion = lpa = rpa = None
```
Comment (larsoner) [inline mne/channels/_standard_montage_utils.py]: This change does not look like reading one versus the other but rather whether or not the nasion/LPA/RPA are defined for the montage. Can you explain a bit?

Comment (christianbrodbeck) [reply mne/channels/_standard_montage_utils.py]: Selecting the right cap happens on line 130/133 in the next hunk below. This here is a secondary change -- the M10 montage has different channel names so inferring the nasion can't be done with the same code. Based in [this](https://www.easycap.de/wp-content/uploads/2018/02/Easycap-Equidistant-Layouts.pdf) mapping (p. 5) the same heuristic could be applied though.

Comment (christianbrodbeck) [reply mne/channels/_standard_montage_utils.py]: Without inferring fiducials:
map-pre
With inferring fiducials:
map-post

Comment (christianbrodbeck) [reply mne/channels/_standard_montage_utils.py]: Is it worth shifting the inferred LPA/RPA to fit the head more precisely?

Comment (larsoner) [reply mne/channels/_standard_montage_utils.py]: The best thing to do would be to figure out where they are supposed to be in the montage, rather than judge based on the plot

Comment (christianbrodbeck) [reply mne/channels/_standard_montage_utils.py]: > The best thing to do would be to figure out where they are supposed to be in the montage, rather than judge based on the plot

I agree, do you know how to do that? I don't have a physical cap... And that would also apply equally to the M1 montage.

Comment (larsoner) [reply mne/channels/_standard_montage_utils.py]: At least for M1 I think that based on a standard 1020 layout definition we should be able to figure it out. It looks like like T3 and T4 being 10% of the way between LPA/RPA and Cz, and Fpz / Oz are 10% up from Nasion/Inion:

http://chgd.umich.edu/wp-content/uploads/2014/06/10-20_system_positioning.pdf

I'd first verify that the spacings in that doc are (at least approximately) accurate for M1 and then go from there.

For M10 we might want to see if the numerical names have readable equivalents (e.g., "10" is "Cz" or whatever) and go from there

Comment (mmagnuski) [reply mne/channels/_standard_montage_utils.py]: The first plot looks much better - that seems to be the exact problem I have in plots in #7193.
@larsoner, @christianbrodbeck - I have digitized positions for this cap, I think, I can take a look.

Comment (mmagnuski) [reply mne/channels/_standard_montage_utils.py]: (ok, the problem is different in the PR, there it is yet another eeg montage. But I should have easycap digitized too)

Comment (larsoner) [reply mne/channels/_standard_montage_utils.py]: That would be helpful if you can check. And we should check that the percentage arc length math works, too. If it doesn't we should figure out how much they differ and choose which one seems correct.

Comment (christianbrodbeck) [reply mne/channels/_standard_montage_utils.py]: Thanks @larsoner! These are the new montages after adding fiducials based on the spherical coordinate system:

map-rad
m10-rad

File: mne/channels/_standard_montage_utils.py [hunk 2]
```diff
@@ -254,15 +235,19 @@ def _read_elc(fname, head_size):
...
-    ch_names, theta, phi = _safe_np_loadtxt(fname, **options)
+def _read_theta_phi_in_degrees(fname, head_size, fid_names=None,
+                               add_fiducials=False):
+    ch_names, theta, phi = _safe_np_loadtxt(fname, skip_header=1,
+                                            dtype=(_str, 'i4', 'i4'))
+    if add_fiducials:
+        assert fid_names is None
+        fid_names = ['Nasion', 'LPA', 'RPA']
+        ch_names.extend(fid_names)
+        theta = np.append(theta, [115, -115, 115])
```
Comment (larsoner) [inline mne/channels/_standard_montage_utils.py]: Can you add a comment where you got this definition for nasion/lpa/rpa in spherical coords? It will help the next time we think about this

**Comment by mmagnuski** [comment]
These look good, if you still need a digitized coordinates I paste them for the M1 cap here (the file is actually a `.csv` file but github doesn't allow droping such files to posts, so I renamed to `.txt`):

[digitization_rotated_coords.txt](https://github.com/mne-tools/mne-python/files/4081121/digitization_rotated_coords.txt)

This was done with photogrammetry from video frames, without control points in physical units, so the units do not mean anything.
I should have the M10 digitized positions somewhere too - so I can drop them too if they'd be useful.

File: mne/channels/_standard_montage_utils.py (hunk 2 above)
Comment (christianbrodbeck) [reply mne/channels/_standard_montage_utils.py]: Added a comment and change log.

**Comment by christianbrodbeck** [comment]
Trying to `Resolve conflicts` but haven't used that feature before, the top says "committing changes -> `christianbrodbeck:easycap`" but the `Commit merge` button looks scary, whill this commmit into the PR or merge the PR into `master`?

**Comment by agramfort** [comment]
Merge with master manually if easier

**Comment by christianbrodbeck** [comment]
Resolving the conflicts is easy enough but I did not want to merge into `master` before it's approved.

**Comment by mmagnuski** [comment]
@christianbrodbeck I'm not sure I understand. Eric wrote it would be useful if I posted the digitized positions in resolving the issue here, so I did.

**Comment by larsoner** [comment]
> will this commmit into the PR or merge the PR into master?

This would just make a "merge commit" in your PR, it would not merge your changes into `master`. Eventually we'll squash and merge as usual.

> >  in that case you have digitized fiducials so this should be a different issue right?

> These look good, if you still need a digitized coordinates I paste them for the M1 cap here... Eric wrote it would be useful if I posted the digitized positions in resolving the issue here, so I did.

Indeed it would be nice to see at least how close the spherical representation is to the digitized one.

I say we merge this once the conflicts are fixed (via the GitHub interface with a merge commit or via rebasing, whatever you find easiest), and then I'm happy to take a quick look at some point at the digitized file (it's not a blocker here). I would say don't bother with the M10 for now unless it's trivial or you are motivated to find it.

**Comment by christianbrodbeck** [comment]
Ah, I made the merge commit.

**Comment by mmagnuski** [comment]
@larsoner @christianbrodbeck 
Sure, I can even take a look at the comparison once this PR is merged.

**Comment by agramfort** [comment]
can you touch examples that use this easycap-M10 file such as

https://mne.tools/dev/auto_examples/visualization/plot_montage.html#sphx-glr-auto-examples-visualization-plot-montage-py

to see how it looks like now?

**Comment by christianbrodbeck** [comment]
@agramfort This example's example [build artifact](https://17594-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_montage.html#sphx-glr-auto-examples-visualization-plot-montage-py) does not seem to contain mayavi images, and a search for "easycap" does not yield any other examples...

**Comment by christianbrodbeck** [comment]
@agramfort This example's example [build artifact](https://17594-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_montage.html#sphx-glr-auto-examples-visualization-plot-montage-py) does not seem to contain mayavi images, and a search for "easycap" does not yield any other examples...

**Comment by mmagnuski** [comment]
@christianbrodbeck You need to change something in an example for it to render. That's what Alex meant with "touch".

**Comment by larsoner** [comment]
I would say that the [new one](https://17597-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_montage.html#sphx-glr-auto-examples-visualization-plot-montage-py) looks better than [master](https://mne.tools/dev/auto_examples/visualization/plot_montage.html#sphx-glr-auto-examples-visualization-plot-montage-py), in that it looks more consistent with the other layouts

**Comment by mmagnuski** [comment]
https://17597-1301584-gh.circle-artifacts.com/0/dev/auto_examples/visualization/plot_montage.html#sphx-glr-auto-examples-visualization-plot-montage-py

@larsoner 
I have one question: why do the channels that are on the circumference of the sphere (3d plot) do not get plotted on the head circle in the topomap? They show within the circle, while the last channel "ring" - the one below the circumference is plotted on the head circle.
Is this behavior expected? I much prefer the circumference to be translated to the head circle (this is also something that eeglab does, IIRC).

**Comment by larsoner** [comment]
In the general case you have to define what you mean by circumference. Relative to what sphere center and what radius?

In spherical montage cases his seems perhaps trivial to define, but that is a specific case of the more general (realistic dig / not true sphere) dig case. At some point we decided that anything at z=0 would be on the head sphere, where z we choose somehow suitably in head coordinates.

Basically I'm open to figuring out how to redefine how we plot things, but it's not a easy as saying "it should be done this way", because it's better if we have something that works for all cases rather than special casing the spherical montage case. So if we are going to change it, we need to understand what we do now and how it should change for all cases.

**Comment by larsoner** [comment]
> where z we choose somehow suitably in head coordinates.

I remembered a detail, which is why this it didn't seem trivial for the cases like this. We want the sensor positions to stay the same regardless when and which channels you pick from the data, i.e., the location of channel X in the full montage should end up being plotted in the same location relative to the spherical head whether or not you've done `raw.pick_channels(['X'])`. Because of how picking works, channel locations in `info['chs']` are lost when picking. So we cannot choose/fit the sphere (origin and radius) to plot relative to using the channel positions, we have to use `info['dig']` instead.

You could say that keeping the EEG locations in `dig` after picking (even if they disappear from `info['chs'][ii]['loc']` could get around this problem. But then you run into a different problem, namely that doing:

- `raw.pick_channels(['X']).set_montage(std).plot_montage()` would look different from doing
- `raw.set_montage(std).pick_channels(['X']).plot_montage()`

because we currently only populate `info['dig']` based on the channels that are actually there.

I guess we could get around this by always populating `info['dig']` with *all* locations from the provided montage even when only a subset of locations are present. Then we could make it so that we always fit with extra+eeg locations in `info['dig']` rather than [only with](https://github.com/mne-tools/mne-python/blob/master/mne/utils/check.py#L593-L602) `extra` points.

I'm not sure if this would have side effects but we could try it. It would at least probably produce the behavior that you were expecting. It's arguably more consistent with what we do in the true-digitization case, because we do not get modify `info['dig']` when picking channels, so even if we are only left with one of the original 60 EEG channels, we will still have 60 (usually actually 61 because ref is in there) locations in `info['dig']`. But it will often break the relationship that originally exists in true files, which is the points in `info['dig']` do correspond exactly to the locations of the (ref plus) channels in the data file.

Modifying the `dig` code is not easy, though, so if this seems like a reasonable approach I can look 
-- I messed with that code recently. I think I'm +0.5 on trying it to see if it helps

**Comment by mmagnuski** [comment]
> In spherical montage cases his seems perhaps trivial to define, but that is a specific case of the more general (realistic dig / not true sphere) dig case. At some point we decided that anything at z=0 would be on the head sphere, where z we choose somehow suitably in head coordinates.

> Basically I'm open to figuring out how to redefine how we plot things, but it's not a easy as saying "it should be done this way", because it's better if we have something that works for all cases rather than special casing the spherical montage case. So if we are going to change it, we need to understand what we do now and how it should change for all cases.

Oh, I didn't mean to sound "it should be done this way". What you describe sounds ok - the user can easily change the coordinates so that the channels are plotted the way they prefer.
I was just wondering why the 3d sphere and channel locations in the rendered example look as if the last channel "ring" was <z but it was still plotted on the head circle in the examples posted by @christianbrodbeck.

> So we cannot choose/fit the sphere (origin and radius) to plot relative to using the channel positions, we have to use info['dig'] instead.

I don't see a problem here actually (but I may be missing something) - the default sphere is in `(0, 0)` and if the center is somewhere else the user can change that using `sphere=` (and the same for radius).

I don't want to pollute this PR with my slightly off-topic questions, we can move the discussion somwehere else - I can for example open a PR to add the info about topomap plotting and `sphere` to the channel positions tutorial (or add this info to the topomap example) and we could continue there.

**Comment by larsoner** [comment]
Sounds good to me

**Comment by christianbrodbeck** [comment]
As far as I am concerned this is ready for merge, I don't think there was anything left to address?

**Comment by larsoner** [comment]
Failures are unrelated `pip` 0.20 errors, thanks @christianbrodbeck
